proto/asgt/v2/dataset_service.proto

```diff
@@ -87,6 +87,8 @@ service DatasetService {
       get: "/v2/datasets/{dataset_name}/trainings/{number_of_trainings}"
     };
   }
+
+  rpc GetConsumerTrainings(ConsumerTrainingsRequest) returns (asgt.v2.type.ConsumerTrainingsResponse);
```

Comment:
I would call this `GetTrainings` and current `GetTrainings` --> `GetDatasetTrainings` 

furthermore, I'd add an HTTP1 path as well 

```
rpc GetTrainings (GetTrainingsRequest) returns (GetTrainingsResponse) {
    option (google.api.http) = {
      get: "/v2/trainings"
      body: "options" // not sure about this name
   };
}

message GetTrainingsRequest {
    ...
    TrainingsOptions options = xyz;
}

message TrainingsOptions {
    int limit = 1; // limit for no of trainings to be returned
}
```

Reply:
Also I think having response message in the `dataset_service.proto` is more fitting since request is here as well

Reply:
I absolutely agree. Just regarding the code you put here, since `GetTrainings` should be a get request, it doesn't really have a post body, so I'd suggest doing something similar to what Miro implemented in his previous trainings call, something along those lines:
```proto
  rpc GetDatasetTrainings (GetDatasetTrainingsRequest) returns (TrainingsResponse) {
    option (google.api.http) = {
      get: "/v2/datasets/{dataset_name}/trainings/{number_of_trainings}"
    };
  }

  rpc GetTrainings(GetTrainingsRequest) returns (TrainingsResponse) {
    option (google.api.http) = {
      get: "/v2/trainings/{number_of_trainings}"
    };
  }

  ...

  message GetTrainingsRequest {
    uint32 number_of_trainings = 1;
  }

  message TrainingsResponse {
    repeated asgt.v2.type.Training trainings = 1;
  }
```

Reply:
the body would map to query parameters... there's an issue with miro's implementation where `get: "/v2/trainings/{number_of_trainings}"` number of trainings is a place for an ID of training and not a limit ... 

see https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L128

Reply:
instead of `"/v2/trainings/{number_of_trainings}"` we want `"/v2/trainings?limit=10"` because then if needed we can do `"/v2/trainings/{training_id}"` to return specific trainings

Reply:
Alright, it makes sense that the request structure should be `/v2/trainings/{training_id}`, in this case `/v2/trainings?limit=10` - I haven't realized this, my bad.

However, after a little bit of digging, I don't think you're right about the `body` being mapped to the query parameters. It doesn't say anywhere actually, nor I could find implementation of `get` with a `body`. What seems like a solution (https://github.com/googleapis/googleapis/blob/0c2e4540e01637d1964d141e68e8cd20619e4c6e/google/api/http.proto#L89) is something like this:
```proto
  rpc GetDatasetTrainings (GetDatasetTrainingsRequest) returns (TrainingsResponse) {
    option (google.api.http) = {
      get: "/v2/datasets/{dataset_name}/trainings"
    };
  }

  rpc GetTrainings(GetTrainingsRequest) returns (TrainingsResponse) {
    option (google.api.http) = {
      get: "/v2/trainings"
    };
  }

  ...

  message GetDatasetTrainingsRequest {
    string dataset_name = 1 [...validation rules];  // maps to the url placeholder `dataset_name`
    int64 limit = 2;    // according to the docs, this should translate into URL query parameter `limit`.
}

  message GetTrainingsRequest {
    int64 limit = 1;   // according to the docs, this should translate into URL query parameter `limit`.
  }
```

The `body` seems to be reserved only for the http requests that actually can contain http body (at least according to this https://github.com/googleapis/googleapis/blob/0c2e4540e01637d1964d141e68e8cd20619e4c6e/google/api/http.proto#L124)